openstreetmap-carto.lua

```diff
@@ -303,134 +574,178 @@ function filter_tags_generic(tags)
         end
     end
 
-   -- Filter out objects that have no tags after deleting
-    if next(tags) == nil then
-        return 1, {}
+    return next(tags) == nil
+end
+
+--- Splits a tag into tags and hstore tags
+-- @return columns, hstore tags
+function split_tags(tags, tag_map)
+    local cols = {tags = {}}
+    for key, value in pairs(tags) do
+        if tag_map[key] then
+            cols[key] = value
+        else
+            cols.tags[key] = value
+        end
     end
+    return cols
+end
+
+phase2_admin_ways = {}
 
-    -- Convert layer to an integer
-    tags['layer'] = layer(tags['layer'])
-    return 0, tags
+-- TODO: Make add_* take object, not object.tags
+function add_point(tags)
```

Comment:
Hi @pnorman,

Looking through these code lines, I noticed you only added the 'layer' key de-duplication in the line type tables, not the 'add_point', 'add_polyon' and 'add_transport_polygon' table polygons, even though these define a 'layer' column as well. Is there any specific reason for this?

Reply:
Nope, I'll go back and look at it.